apps/governance/src/routes/proposals/components/proposal-referral-program-details/proposal-referral-program-details.tsx

```diff
@@ -0,0 +1,233 @@
+import { useTranslation } from 'react-i18next';
+import type { ProposalQuery } from '../../proposal/__generated__/Proposal';
+import {
+  KeyValueTable,
+  KeyValueTableRow,
+  RoundedWrapper,
+  Tooltip,
+} from '@vegaprotocol/ui-toolkit';
+import { formatNumber } from '../../../../lib/format-number';
+import {
+  formatDateWithLocalTimezone,
+  formatNumberPercentage,
+  toBigNum,
+} from '@vegaprotocol/utils';
+import BigNumber from 'bignumber.js';
+import { useAppState } from '../../../../contexts/app-state/app-state-context';
+
+interface ProposalReferralProgramDetailsProps {
+  proposal: ProposalQuery['proposal'];
+}
+
+export const formatEndOfProgramTimestamp = (value: string) => {
+  return formatDateWithLocalTimezone(new Date(value));
+};
```

Comment:
**nitpick (non-blocking)**:  These could be more defensive. I realise Typescript *feels* like it means that these values can never be `null` or `undefined` but changes between versions or just plain old data node bugs can mean that they end up being unexpected values that will throw errors. `try`/`catch`ing or  testing that a string is valid would be good... but at least some of the underlying functions probably already do, so I've marked it as non-blocking.